src/main/java/tdl/client/runner/ChallengeSession.java

```diff
@@ -13,57 +12,27 @@
     private int port;
     private String journeyId;
     private boolean useColours;
-    private final String username;
     private ImplementationRunner implementationRunner;
     private AuditStream auditStream;
     private RecordingSystem recordingSystem;
     private ChallengeServerClient challengeServerClient;
     private ActionProvider userInputCallback;
 
-    public static ChallengeSession forUsername(@SuppressWarnings("SameParameterValue") String username) {
-        return new ChallengeSession(username);
+    public static ChallengeSession forRunner(ImplementationRunner implementationRunner) {
+        return new ChallengeSession(implementationRunner);
     }
 
-    private ChallengeSession(String username) {
-        this.username = username;
-        this.port = 8222;
-        this.useColours = true;
-        this.recordingSystem = new RecordingSystem(true);
-        this.auditStream = new StdoutAuditStream();
+    private ChallengeSession(ImplementationRunner runner) {
+        this.implementationRunner = runner;
     }
 
-    public ChallengeSession withServerHostname(@SuppressWarnings("SameParameterValue") String hostname) {
-        this.hostname = hostname;
-        return this;
-    }
-
-    public ChallengeSession withPort(int port) {
-        this.port = port;
-        return this;
-    }
-
-    public ChallengeSession withJourneyId(String journeyId) {
-        this.journeyId = journeyId;
-        return this;
-    }
-
-    public ChallengeSession withColours(boolean useColours) {
-        this.useColours = useColours;
-        return this;
-    }
-
-    public ChallengeSession withAuditStream(AuditStream auditStream) {
-        this.auditStream = auditStream;
-        return this;
-    }
-
-    public ChallengeSession withImplementationRunner(ImplementationRunner implementationRunner) {
-        this.implementationRunner = implementationRunner;
-        return this;
-    }
-
-    public ChallengeSession withRecordingSystemOn(boolean recordingSystemOn) {
-        this.recordingSystem = new RecordingSystem(recordingSystemOn);
+    public ChallengeSession withConfig(ChallengeSessionConfig config) {
+        this.hostname = config.getHostname();
```

Comment:
We just need to store the config itself and remove the other variables.

Reply:
Who owns the RecordingSystem member variable?

Reply:
`ChallengeSession`